auxin_cli/src/main.rs

```diff
@@ -163,6 +165,91 @@ impl SignalPrekeyData {
 	}
 }
 
+/// Does last steps to fully register a signal account
+async fn fix_accounts<R>(client: &reqwest::Client, account: &SignalAccount<R>) -> Result<()> {
+	// TODO(Diana): This seems to basically just be
+	// AuxinApp::retrieve_profile but on ourselves
+	// But we can't use AuxinApp here
+	let uuid = account.aci().uuid();
+	let version = account.profile_key().version(uuid);
+	let url = format!("https://chat.signal.org/v1/profile/{uuid}/{version}",);
+	let res = client
+		.get(url)
+		.basic_auth(account.aci().uuid(), Some(account.password()))
+		.send()
+		.await?;
+	// TODO(Diana): We don't currently care about the response
+	let _profile: auxin::profile::ProfileResponse = res.json().await?;
+
+	// Construct a default profile for our new user.
+	let profile = auxin::profile::ProfileConfig {
+		about: None,
+		mobilecoin_address: None,
+		mood_emoji: None,
+		name: auxin::profile_cipher::ProfileName {
+			given_name: String::from(account.phone()),
+			family_name: None,
+		},
+		avatar_file: None,
+	};
+	// Terrible hack to build the profile request without duplicating all the code.
+	let profile = auxin::profile::build_set_profile_request(
+		profile,
+		&auxin::LocalIdentity {
+			address: auxin::address::AuxinDeviceAddress {
+				address: AuxinAddress::Both(
+					account.phone().into(),
+					uuid::Uuid::parse_str(account.aci().uuid())?,
+				),
+				device_id: 1,
+			},
+			password: account.password().into(),
+			profile_key: *account.profile_key().as_bytes(),
+			identity_keys: libsignal_protocol::IdentityKeyPair::new(
+				libsignal_protocol::IdentityKey::new(PublicKey::from_djb_public_key_bytes(
+					account.aci().identity().public(),
+				)?),
+				libsignal_protocol::PrivateKey::deserialize(account.aci().identity().private())?,
+			),
+			reg_id: account.registration_id() as u32,
+		},
+		&mut rand::thread_rng(),
+	)?;
+	client
+		.put("https://chat.signal.org/v1/profile")
```

Comment:
These signal endpoint uris are moving targets - so Maybe we can have some sort of config enum that contains the endpoints data instead of hardcoding it.

Reply:
Yes, that was to-do list item #1 when I was planning to refactor Auxin. It really needs to be defined declarative somewhere rather than just having this string literal all over the place. Otherwise we can't use Signal testnet.

That might be out of the scope of this PR though.